spec/models/user_spec.rb

```diff
@@ -1,47 +1,58 @@
-require 'support/notifications_service'
-
 describe User do
-  include_examples 'notifications service'
-
   context 'associations' do
     it { should have_many(:ips) }
     it { should belong_to(:organisation) }
   end
 
-  describe '#update_details' do
-    let(:user) { create(:user, password: 'password') }
-    context 'when passwords match' do
-      let(:params) { { password: '123456', password_confirmation: '123456', name: 'bob' } }
+  context 'validations' do
+    it { should validate_presence_of(:name).on(:update) }
 
-      it 'should set the users password' do
-        expect { user.update_details(params) }.to change(user, :password)
-        expect(user.errors.empty?).to eq(true)
+    describe "password and password confirmation must match" do
+      let(:user) { create(:user, :confirmed, name: "Old name") }
+      before do
+        user.update(params)
       end
 
-      context 'Name validation' do
-        it { should validate_presence_of(:name).on(:update) }
+      context 'when passwords match' do
+        let(:params) { { password: 'new_password', password_confirmation: 'new_password', name: "New name" } }
+
+        it 'should set the users password' do
+          expect(user.errors.empty?).to eq(true)
+          expect(user.name).to eq("New name")
+        end
       end
-    end
 
-    context 'when passwords do not match' do
-      let(:params) { { password: '123456', password_confirmation: '1234567' } }
-      it 'should not set the users password' do
-        expect { user.update_details(params) }.to_not change(user, :password)
-        expect(user.errors.empty?).to eq(false)
-        expect(user.errors.full_messages).to eq(['Password must match confirmation'])
+      context 'when passwords do not match' do
+        let(:params) { { password: 'new_password', password_confirmation: 'other_password', name: "New name" } }
+
+        it 'should not set the users password' do
+          expect(user.reload.name).to eq("Old name")
+          expect(user.errors.empty?).to eq(false)
```

Comment:
I would get rid of these `.empty?` checks - think the lines below cover it enough.